scripts/generate_ipfs_hashes.py

```diff
@@ -57,133 +56,105 @@
-    ignore_patterns = config.get("fingerprint_ignore_patterns", [])
-    if package_type != "agents":
-        # hash inner components
-        fingerprints_dict = _compute_fingerprint(Path(target_dir), ignore_patterns)
-        # confirm ipfs only generates same hash:
-        for file_name, ipfs_hash in fingerprints_dict.items():
-            path = os.path.join(target_dir, file_name)
-            ipfsho_hash = ipfs_hash_only.get(path)
-            if ipfsho_hash != ipfs_hash:
-                print("WARNING, hashes don't match for: {}".format(path))
-
-        # update fingerprints
-        file_name = package_type[:-1] + ".yaml"
-        yaml_path = os.path.join(target_dir, file_name)
-        file = open(yaml_path, mode="r")
-
-        # read all lines at once
-        whole_file = file.read()
-
-        # close the file
-        file.close()
-
-        file = open(yaml_path, mode="r")
-
-        # find and replace
-        # TODO this can be simplified after https://github.com/fetchai/agents-aea/issues/932
-        existing = ""
-        fingerprint_block = False
-        for line in file:
-            if line.find("fingerprint:") == 0:
-                existing += line
-                fingerprint_block = True
-            elif fingerprint_block:
-                if line.find("  ") == 0:
-                    # still inside fingerprint block
-                    existing += line
-                else:
-                    # fingerprint block has ended
-                    break
-
-        if len(fingerprints_dict) > 0:
-            replacement = "fingerprint:\n"
-            ordered_fingerprints_dict = collections.OrderedDict(
-                sorted(fingerprints_dict.items())
-            )
-            for file_name, ipfs_hash in ordered_fingerprints_dict.items():
-                replacement += "  " + file_name + ": " + ipfs_hash + "\n"
-        else:
-            replacement = "fingerprint: {}\n"
-        whole_file = whole_file.replace(existing, replacement)
+    assert config.directory is not None
+    configuration_filepath = config.directory / config.default_configuration_filename
+    yaml_dump(config.ordered_json, configuration_filepath.open("w"))
 
-        # close the file
-        file.close()
 
-        # update fingerprints
-        with open(yaml_path, "w") as f:
-            f.write(whole_file)
+def ipfs_hashing(
+    client: ipfshttpclient.Client,
+    configuration: PackageConfiguration,
+    package_type: ConfigurationType,
+):
+    """
+    Hashes a package and its components.
+
+    :param client: a connected IPFS client.
+    :param configuration: the package configuration.
+    :param package_type: the package type.
+    :return: the hash of the whole package.
+    """
+    print("Processing package {} of type {}".format(configuration.name, package_type))
 
     # hash again to get outer hash (this time all files):
     # TODO we still need to ignore some files
```

Comment:
the outer hash should not ignore anything.

Reply:
why? I think it should at least ignore the files in the "fingerprint_ignore" stated in the packages configuation. Anyway, It is not implemented yet.

Reply:
We could use `.ipfsignore` (#3643) to be added "on-the-fly" before the `ipfs add`. Or, we could think of having such file in every package (optionally).

Probably a topic for a new PR :-)

Reply:
No it must absolutely not ignore anything. The outer hash is to content address the package folder as is, it must hash all files otherwise the package can not be fetched in its entirety. Also as a matter of separation of concerns this PR should not change any of the hashing. Any changes can be proposed in a separate issue and discussed there.